[Accueil](../../index.ipynb) > [Sommaire de Terminale](../index.ipynb) > [3.4 Langage SQL](langage_SQL.ipynb)

# Une base de données pour une maternité

Cet exercice est inspiré de l'exercice 1 du sujet 2022 d'Amérique du Sud, jour1

Voici une représentation des relations de la base.

![](img/schema_maternite.png)

Dans la relation *Naissances*:
- La clé étrangère *idMere* fait référence à la clef primaire *numPatiente* de la relation *Patientes*;
- La clé étrangère *acc* fait référence à la clef primaire *idAcc* de la relation *TypesAccouchement*. Il permet de savoir si la patiente à accouché par voie naturelle ou par césarienne;
- l’attribut *rang* indique le rang de naissance du bébé dans le mois. Il recommence donc à 1 au début de chaque nouveau mois;
- l’attribut *poids* est exprimé en grammes et l’attribut *taille* en centimètres.


In [ ]:
import sqlite3 as sql
%load_ext sql
%sql sqlite:///dbs/maternite.db

## Liste des relations de la base de données.

In [ ]:
%sql SELECT name FROM sqlite_master  WHERE type='table' AND name != 'sqlite_sequence';

## Le contenu de la relation *Naissances*.

In [ ]:
%sql SELECT * FROM Naissances;

## Le contenu de la relation *Patientes*.

In [57]:
%sql SELECT * FROM Patientes;

 * sqlite:///dbs/maternite.db
Done.


[(13857, 'Berthelot', 'Michelle', 'Aigrefeuille d’Aunis'),
 (13858, 'Samson', 'Marine', 'Nieul sur Mer'),
 (13859, 'Perrin', 'Patricia', 'La Rochelle'),
 (13860, 'Fernandez', 'Claire', 'Aytré'),
 (13861, 'Baugé', 'Gaëlle', 'Lagord')]

## Le contenu de la relation *TypesAccouchement*.

In [ ]:
%sql SELECT * FROM TypesAccouchement;

## Questions

1. Indiquer, pour chacune des propositions suivantes, si elles peuvent être ou non clé primaire de la relation *Naissances*. Justifier chaque réponse.

    1. idMere
    2. (date, rang)
    3. (poids, taille)
    
2. Pourquoi la requête ci-dessous provoque-t-elle une erreur?

    DELETE FROM Patientes WHERE numPatiente = 13858;
    
3. Donner en langage SQL la requête d’insertion qui permet d’inscrire, avec le numéro 13862, la patiente nommée Ninette Bélanger résidant à la Rochelle.

In [ ]:
%sql

4. Dans l’extrait de la table fourni, on constate qu’une erreur a été commise lors de la saisie du prénom du bébé de madame Fernandez. La bonne orthographe est Laurette.
   Donner en langage SQL l’instruction à exécuter pour corriger le prénom du bébé.

In [ ]:
%sql

5. Ecrire une requête SQL qui renvoie le poids moyen des bébés nés par césarienne. On utilisera la fonction d'agrégation AVG(attr)

In [ ]:
%sql

6. Afficher les noms et prénoms des mères qui ont accouché par voie naturelle. (La question d'origine était différente) 

In [ ]:
%sql #utiliser le caractère \ pour faire des retours à la ligne

In [ ]:
%sql \
SELECT DISTINCT Patientes.nom AS "Nom mère", Patientes.prenom AS "Prénom mère" \
FROM Naissances \
JOIN Patientes ON Naissances.idMere=Patientes.numPatiente\
WHERE Naissances.acc=1;

[Accueil](../../index.ipynb) > [Sommaire de Terminale](../index.ipynb) > [3.4 Langage SQL](langage_SQL.ipynb)